components/common/src/command/package/install.rs

```diff
@@ -816,8 +816,9 @@ impl<'a> InstallTask<'a> {
         // Canonicalize both paths to ensure that there aren't any symlinks when comparing them
         // later.
         let artifact_path = artifact_path.canonicalize()?;
-        let cache_path = self.cached_artifact_path(ident).canonicalize()?;
         fs::create_dir_all(self.artifact_cache_path)?;
+        let artifact_cache_path = self.artifact_cache_path.canonicalize()?;
+        let cache_path = artifact_cache_path.join(ident.archive_name());
```

Comment:
I think you're going to need to rebase your PR so you don't accidentally revert https://github.com/habitat-sh/habitat/pull/4864

Let me know if you need any help with that

Reply:
Yeah. I have bad rebase on this and the other open PR. Chris is going to help me with it today. Thanks for pointing it out Jon!